thirdparty/cmdconfig/commands/fnevalcmd_test.go

```diff
@@ -211,32 +211,72 @@ apiVersion: v1
 		},
 		{
 			name: "log steps",
-			args: []string{"eval", "dir", "--log-steps"},
+			args: []string{"eval", "dir", "--log-steps", "--image", "foo:bar"},
 			path: "dir",
 			expectedStruct: &runfn.RunFns{
 				Path:     "dir",
 				LogSteps: true,
 				Env:      []string{},
 			},
+			expected: `
+metadata:
+  name: function-input
+  annotations:
+    config.kubernetes.io/function: |
```

Comment:
what do we still have this annotation?

Reply:
Most test cases are not touched in this PR. I will update them.

Reply:
Just noticed that you are talking about the `fnevalcmd`. The logic here is the `fnevalcmd` will convert the imperative command line arguments to a ConfigMap like this so it can consistent with declarative function run and `runfn` will only need to handle a unique format. If we change this method we need to rewrite almost all codes in the `fnevalcmd` and `runfn`.

Reply:
I didn't parse this. Ultimately, this annotation needs to go away right?

Reply:
Some context, for now, the approach has been to do minimal changes so that we leverage the good parts (primarily the testing mileage this code has seen) for now.

Reply:
> I didn't parse this. Ultimately, this annotation needs to go away right?

These annotations are used to pass the function specs from `fnevalcmd` to `runfn`. They are left so we can reuse most of the tests in `runfn_test.go` now.

Reply:
You need to make the intentions clear to the reviewer using TODOs if there's going to be a cleanup. This needs to be done proactively.

My question: you'll delete these in a follow up PR?  This annotation is part of the surface area that no longer makes sense in v1.

Reply:
Yes we will have another PR for this because it will be a huge one. Add a TODO for `getContainerFunctions`.